projects/file-parsing/README.md

```diff
@@ -0,0 +1,56 @@
+<!--forhugo
++++
+title="File Parsing"
++++
+forhugo-->
+
+In this project we'll practice parsing data from files in different formats. Often times it's convenient to write a small program to understand or process some data, and being comfortable quickly doing so can be very helpful.
+
+Timebox: 2 days
+
+## Objectives:
+
+- Write code to parse data in non-standard formats.
+- Become comfortable leveraging libraries to parse standard formats in slightly non-standard ways.
+
+## Project
+
+The `examples` sub-directory in this directory contains a number of data files, each of which contains the same data in a different format. Descriptions of each format can be found below. Each data file contains a data-set of player names, and their high scores, for some game.
+
+We are going to write a program to analyse the data, and print out the names of the players with the highest and lowest scores.
+
+### Data formats
+
+### JSON
+
+json.txt contains an array of JSON objects, each with a "name" and "high_score" key.
+
+### Repeated JSON
+
+repeated-json.txt contains lines of data stored in JSON format. Each line contains exactly one record, stored as an object. Lines starting with # are comments and should be ignored.
+
+### CSV
+
+data.csv is a standard CSV file. The format is well-documented online, and there are many libraries which support parsing it.
+
+### Custom Binary
+
+There are two files in a custom binary serialisation format. The format is as follows:
+* First two bytes of the file indicate endianness of numbers. If the bytes are FE FF, numbers in the file are stored in big endian byte order. If the bytes are FF FE, numbers in the file are stored in little endian byte order.
+* Each record contains exactly four bytes representing the score as a signed 32-bit integer, in the above described endian format, then the name of the player stored in UTF-8 which may not contain a null character, followed by a null terminating character.
```

Comment:
you may also need to explain UTF 8 and terminating characters here. i don't think we can assume people have had the experience of wrangling C style strings

Reply:
Good call - added some definitions, and then a section to consider trade-offs of different encodings of variable length data (see also https://github.com/CodeYourFuture/immersive-go-course/issues/112)